odo/backends/aws.py

```diff
@@ -219,15 +220,44 @@ def anything_to_s3_text(s3, o, **kwargs):
     return append(s3, convert(Temp(s3.subtype), o, **kwargs), **kwargs)
 
 
+@contextmanager
+def start_multipart_upload_operation(s3):
+    try:
+        multipart_upload = s3.object.bucket.initiate_multipart_upload(s3.key)
+        yield multipart_upload
+    except Exception as exception:
+        multipart_upload.cancel_multipart_upload()
+        raise exception
```

Comment:
just reraise here with `raise` instead of `raise exception`:

```python
try:
    ...
except Exception:
    ...
    raise
```

this indicates that you don't intend to handle it as well as removes an unecessary stack frame from the error message